maps/randomvaults/spessmart.dm

```diff
@@ -241,6 +243,8 @@ var/list/clothing_prices = list()	//gets filled on initialize()
 
 /obj/effect/spessmart_entrance/Crossed(atom/movable/AM)
 	if(isliving(AM))
+		var/area/vault/supermarket/shop/A = get_area(src)
```

Comment:
So I can access the `customer_has_entered` var of the area that the entrance marker is inside, and set it to `TRUE` when someone crosses the entrance.

The only way someone can possibly get inside is by going through the entrance, or by breaking one of the front windows which would cause a lockdown anyway.